text/0000-inline-asm.md

```diff
@@ -233,9 +234,9 @@ unsafe {
     asm!(
         "cpuid",
         // EAX 4 selects the "Deterministic Cache Parameters" CPUID leaf
-        inout("eax") 4 => _,
+        inout("eax") 4u64 => _,
         // ECX 0 selects the L0 cache information.
-        inout("ecx") 0 => ecx,
+        inout("ecx") 0u64 => ecx,
```

Comment:
shouldn't these be `u32`?

Reply:
The issue here was that the two parts of `inout` must have the same type. So either both are `u32` or both are `u64`. However 0 was defaulting to `i32` which causes this example to fail to compile.

Reply:
ok, though this seems like an indication that requiring matching types for `inout` in all cases might not be the best idea, as long as the inputs and outputs can still be allocated to the same registers.

Also, as far as I know, `cpuid`'s inputs and outputs are all 32-bit values, so using u32 should be appropriate here (though I haven't checked this particular subcommand so might be wrong).

Reply:
This limitation comes from LLVM, there's nothing I can do about it.

Reply:
> This limitation comes from LLVM, there's nothing I can do about it.

ah, oh well.

Reply:
Could we fix this by automatically generation approriate LLIR before/after the asm block? Not saying we should do this now, but this could be an option in the future as we start supporting more flexible backends.